Test/Framework/CmdlineOptions.hs

```diff
@@ -181,6 +183,10 @@ optionDescriptions =
     , Option []        ["timeout-is-success"]
              (NoArg (\o -> Right $ o { opts_timeoutIsSuccess = True }))
              "Do not regard a test timeout as an error."
+    , Option []        ["repeat"]
+             (ReqArg (\s o -> parseRead "--repeat" s >>= \(i::Int) ->
+                              Right $ o { opts_repeat = i}) "NUMBER")
+             "Execute all tests NUMBER times"
```

Comment:
Do you really execute *all* tests NUMBER times? It should only repeat the tests given by the commandline arguments (which defaults to all if you don't specify any arguments).